src/EventStore.Core/Services/Storage/ReaderIndex/AllReader.cs

```diff
@@ -258,5 +318,24 @@ public class AllReader : IAllReader {
 			       || (rec.RecordType == LogRecordType.Prepare &&
 			           ((PrepareLogRecord)rec).Flags.HasAnyOf(PrepareFlags.IsCommitted));
 		}
+
+		private struct ReadAllEventsRawResult {
```

Comment:
We don't need `ReadAllEventsRawResult` as it's just there to carry the `IsEndOfStream` property to the caller. We can also get rid of the `IndexReadAllFilteredResult` and add the `IsEndOfStream` property to the `IndexReadAllResult`.

Reply:
I agree, good idea